sql/types/jsontests/json_tests.go

```diff
@@ -12,305 +12,116 @@
 // limitations under the License.
 
-package types
+package jsontests
 
 import (
 	"context"
 	"fmt"
-	"reflect"
 	"testing"
 
 	"github.com/dolthub/go-mysql-server/sql"
-	// _ "github.com/dolthub/go-mysql-server/sql/variables"
-	"github.com/dolthub/vitess/go/vt/proto/query"
+	"github.com/dolthub/go-mysql-server/sql/types"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestJsonCompare(t *testing.T) {
-	tests := []struct {
-		left  string
-		right string
-		cmp   int
-	}{
-		// type precedence hierarchy: BOOLEAN, ARRAY, OBJECT, STRING, DOUBLE, NULL
-		{`true`, `[0]`, 1},
-		{`[0]`, `{"a": 0}`, 1},
-		{`{"a": 0}`, `"a"`, 1},
-		{`"a"`, `0`, 1},
-		{`0`, `null`, 1},
-
-		// json null
-		{`null`, `0`, -1},
-		{`0`, `null`, 1},
-		{`null`, `null`, 0},
-
-		// boolean
-		{`true`, `false`, 1},
-		{`true`, `true`, 0},
-		{`false`, `false`, 0},
-
-		// strings
-		{`"A"`, `"B"`, -1},
-		{`"A"`, `"A"`, 0},
-		{`"C"`, `"B"`, 1},
-
-		// numbers
-		{`0`, `0.0`, 0},
-		{`0`, `-1`, 1},
-		{`0`, `3.14`, -1},
-
-		// arrays
-		{`[1,2]`, `[1,2]`, 0},
-		{`[1,9]`, `[1,2]`, 1},
-		{`[1,2]`, `[1,2,3]`, -1},
-
-		// objects
-		{`{"a": 0}`, `{"a": 0}`, 0},
-		// deterministic object ordering with arbitrary rules
-		{`{"a": 1}`, `{"a": 0}`, 1},                 // 1 > 0
-		{`{"a": 0}`, `{"a": 0, "b": 1}`, -1},        // longer
-		{`{"a": 0, "c": 2}`, `{"a": 0, "b": 1}`, 1}, // "c" > "b"
-
-		// nested
-		{
-			left:  `{"one": ["x", "y", "z"], "two": { "a": 0, "b": 1}, "three": false, "four": null, "five": " "}`,
-			right: `{"one": ["x", "y", "z"], "two": { "a": 0, "b": 1}, "three": false, "four": null, "five": " "}`,
-			cmp:   0,
-		},
-		{
-			left:  `{"one": ["x", "y"],      "two": { "a": 0, "b": 1}, "three": false, "four": null, "five": " "}`,
-			right: `{"one": ["x", "y", "z"], "two": { "a": 0, "b": 1}, "three": false, "four": null, "five": " "}`,
-			cmp:   -1,
-		},
-	}
-
-	for _, test := range tests {
-		name := fmt.Sprintf("%v_%v__%d", test.left, test.right, test.cmp)
-		t.Run(name, func(t *testing.T) {
-			cmp, err := JSON.Compare(
-				MustJSON(test.left),
-				MustJSON(test.right),
-			)
-			require.NoError(t, err)
-			assert.Equal(t, test.cmp, cmp)
-		})
-	}
-}
-
-func TestJsonCompareNulls(t *testing.T) {
-	tests := []struct {
-		left  interface{}
-		right interface{}
-		cmp   int
-	}{
-		{nil, MustJSON(`{"key": "value"}`), 1},
-		{MustJSON(`{"key": "value"}`), nil, -1},
-		{nil, nil, 0},
-		{nil, MustJSON(`null`), 1},
-		{MustJSON(`null`), nil, -1},
-	}
-
-	for _, test := range tests {
-		name := fmt.Sprintf("%v_%v__%d", test.left, test.right, test.cmp)
-		t.Run(name, func(t *testing.T) {
-			cmp, err := JSON.Compare(test.left, test.right)
-			require.NoError(t, err)
-			assert.Equal(t, test.cmp, cmp)
-		})
+func ConvertToJson(t *testing.T, val interface{}) types.MutableJSON {
+	if val == nil {
+		return nil
 	}
+	val, inRange, err := types.JSON.Convert(val)
+	require.NoError(t, err)
```

Comment:
Does this file need to end in `_test.go` for it to be only included in the test closure? Or... will this end up pulling in test dependencies, like `require` into the main dolt executable? I think I remember you saying the partitioning was on a file level, so this file wouldn't be pulled in if it's only used by tests, but wanted to double check.

Reply:
We actually need this to *not* be included in test closure, or else we won't be able to import it in Dolt tests.

The package level partitioning is why this is pulled into its own package: since only the test closure depends on this package, it won't get pulled into the main dolt executable.